### PR TITLE
[Fix] Some of the allowed tags is not working on email notifications set via Email Templates options

### PR DESCRIPTION
**Pull request by mebishalnapit**
This PR includes:

* Fix the **{site_name}**, **{site_url}**, and **{site_description}** smart tags for email delivery change to required contents for **New Question**, **New Answer**, and **New Comment** email templates
* Fix the **{answer_excerpt}** smart tags for email delivery change to required contents for **Edit Answer** email templates
* Fix the **More email options** link available in the **Email Templates** option

### CODE REVIEW

File: includes/functions.php [hunk 1]
```diff
@@ -2493,3 +2493,27 @@ function ap_get_current_timestamp() {
...
+ * Merges user defined array arguments into the defaults array,
+ * which works with arrays only.
+ *
+ * @param array $args     Value to be merged with the defaults array.
+ * @param array $defaults Array that serves as the defaults.
+ *
+ * @return array Merged user defined values with the default array.
+ * @since 4.4.0
+ */
+function ap_parse_args( $args, $defaults ) {
```
Comment (rahularyan) [inline includes/functions.php:2507]: @mebishalnapit Add type casting for the two arguments.

**Review by rahularyan — CHANGES_REQUESTED** [review]

File: includes/functions.php (hunk 1 above)
Comment (mebishalnapit) [reply includes/functions.php]: @rahularyan Added.

File: includes/functions.php [hunk 2]
```diff
@@ -2493,3 +2493,27 @@ function ap_get_current_timestamp() {
...
+ * which works with arrays only.
+ *
+ * @param array $args     Value to be merged with the defaults array.
+ * @param array $defaults Array that serves as the defaults.
+ *
+ * @return array Merged user defined values with the default array.
+ * @since 4.4.0
+ */
+function ap_parse_args( array $args, array $defaults ) {
+	$new_args = (array) $defaults;
```
Comment (rahularyan) [inline includes/functions.php:2508]: @mebishalnapit You don't need this type casting.

File: includes/functions.php [hunk 3]
```diff
@@ -2493,3 +2493,27 @@ function ap_get_current_timestamp() {
...
+ * Merges user defined array arguments into the defaults array,
+ * which works with arrays only.
+ *
+ * @param array $args     Value to be merged with the defaults array.
+ * @param array $defaults Array that serves as the defaults.
+ *
+ * @return array Merged user defined values with the default array.
+ * @since 4.4.0
+ */
+function ap_parse_args( array $args, array $defaults ) {
```
Comment (rahularyan) [inline includes/functions.php:2507]: @mebishalnapit May be you need default value for `$defaults` to an empty array?

**Review by rahularyan — CHANGES_REQUESTED** [review]

File: includes/functions.php (hunk 2 above)
Comment (mebishalnapit) [reply includes/functions.php]: @rahularyan Removed.

File: includes/functions.php (hunk 3 above)
Comment (mebishalnapit) [reply includes/functions.php]: @rahularyan Modified argument to an empty array.

**Pull request merged**